pkg/agent/runner.go

```diff
@@ -1551,6 +1570,35 @@ func (r *Runner) doVMUpdate(
 	return &target, nil
 }
 
+func (r *Runner) recordResourceChange(current, target api.Resources, metrics resourceChangePair) {
+	abs := current.AbsDiff(target)
+
+	// Add CPU
+	if abs.VCPU != 0 {
+		direction := "up"
+		if target.VCPU < current.VCPU {
+			direction = "down"
+		}
+
+		metrics.cpu.WithLabelValues(direction).Add(abs.VCPU.AsFloat64())
+	}
+
+	// Add memory
+	if abs.Mem != 0 {
+		direction := "up"
+		if target.Mem < current.Mem {
+			direction = "down"
+		}
+
+		// Avoid floating-point inaccuracy.
+		byteTotal := r.vm.Mem.SlotSize.Value() * int64(abs.Mem)
+		mib := int64(1 << 20)
```

Comment:
constant in the package?

why not wrap/convert to floats here just to make the math part on the next line easier to read?

Reply:
> constant in the package?

honestly I think it's easier to understand with the variable closer.

> why not wrap/convert to floats here just to make the math part on the next line easier to read?

So the next line very much _could_ be `float64(byteTotal)/float64(mib)` (which we could wrap with floats beforehand). But floating point math is... always iffy, and so IMO it's worthwhile here to split up `byteTotal/mib` into the integer and fractional parts .